src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Input/Commands/AddClassProjectCommandTests.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Input;
+using Microsoft.VisualStudio.Mocks;
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    [ProjectSystemTrait]
+    public class AddClassProjectCommandTests
+    {
+        [Fact]
+        public void Constructor_NullAsProjectTree_ThrowsArgumentNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => new AddClassProjectCommand(null, IUnconfiguredProjectVsServicesFactory.Create(),
+                SVsServiceProviderFactory.Create()));
+        }
+
+        [Fact]
+        public void Constructor_NullAsProjectVsService_ThrowsArgumentNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => new AddClassProjectCommand(IPhysicalProjectTreeFactory.Create(), null,
+                SVsServiceProviderFactory.Create()));
+        }
+
+        [Fact]
+        public void Constructor_NullAsSVsServiceProvider_ThrowsArgumentNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => new AddClassProjectCommand(IPhysicalProjectTreeFactory.Create(),
+                IUnconfiguredProjectVsServicesFactory.Create(), null));
+        }
+
+        [Fact]
+        public void GetCommandStatusAsync_NullAsNodes_ThrowsArgumentNull()
+        {
+            var command = CreateInstance();
+
+            Assert.Throws<ArgumentNullException>("nodes", () =>
+            {
+                command.GetCommandStatusAsync((IImmutableSet<IProjectTree>)null, GetCommandId(), true, "commandText", CommandStatus.Enabled);
+            });
+        }
+
+        [Fact]
+        public async Task GetCommandStatusAsync_UnrecognizedCommandIdAsCommandId_ReturnsUnhandled()
+        {
+            var command = CreateInstance();
+
+            var tree = ProjectTreeParser.Parse(@"
+Root (flags: {ProjectRoot})
+");
+
+            var nodes = ImmutableHashSet.Create(tree);
+
+#pragma warning disable RS0003 // Do not directly await a Task (see https://github.com/dotnet/roslyn/issues/6770)
```

Comment:
This analyzer shouldn't be firing in this project since it's a test project. Can you double check if the project has a <NonShipping> attribute mentioned?